main.md

```diff
@@ -92,6 +92,40 @@ It is a good practice to make a new branch for every new PR you make. Also,name
 - Then edit the ‘pick’ to ‘squash’ in front of all those commits which you want to squash.
 - Commit your new squashed commits.
 
+
+## Removing a commit from in between the commit history
+
+   If you need to delete more than just the last commit use rebase.
+  
+  Example-
+
+  commit 1	     2c6a45b	  Adding public method to access protected method(HEAD)	
+  
+  commit 2	     ae45fab	  Updates to database interface	
+  
+  commit 3	     77b9b82	  Improving database interface	
+
+  Using the git log above we want to remove the following commit; 2 (ae45fab).
+  
+  `$ git rebase --onto repair~2 repair~1 repair`
+  
+  - This command will delete commit 2
+  
+   Generalized Format for writing command is as follows:
+   
+  `$ git rebase --onto <branch name>~<first commit number to remove> <branch name>~<first commit to be kept> <branch name>`
+  
+   -Use rebase tool to rebase a series of commits onto the HEAD they were originally based on instead of moving them to another one.
+   
+   -Then give branch name along with first commit to be removed.
```

Comment:
add a space btw - and Use, check all the statements below and rectify those

Reply:
okay